app/src/main/java/tech/relaycorp/gateway/domain/LocalConfig.kt

```diff
@@ -0,0 +1,73 @@
+package tech.relaycorp.gateway.domain
+
+import tech.relaycorp.gateway.data.disk.SensitiveStore
+import tech.relaycorp.relaynet.issueGatewayCertificate
+import tech.relaycorp.relaynet.wrappers.x509.Certificate
+import java.security.KeyFactory
+import java.security.KeyPair
+import java.security.KeyPairGenerator
+import java.security.PrivateKey
+import java.security.interfaces.RSAPrivateCrtKey
+import java.security.spec.EncodedKeySpec
+import java.security.spec.PKCS8EncodedKeySpec
+import java.security.spec.RSAPublicKeySpec
+import java.time.ZonedDateTime
+import javax.inject.Inject
+
+class LocalConfig
+@Inject constructor(
+    private val sensitiveStore: SensitiveStore
+) {
+
+    suspend fun getKeyPair() =
+        sensitiveStore.read(PRIVATE_KEY_FILE_NAME)
+            ?.toPrivateKey()
+            ?.toKeyPair()
+            ?: generateKeyPair().also {
+                sensitiveStore.store(PRIVATE_KEY_FILE_NAME, it.private.encoded)
+            }
+
+    suspend fun getCertificate() =
+        sensitiveStore.read(CERTIFICATE_FILE_NAME)
+            ?.let { Certificate.deserialize(it) }
+            ?: generateGatewayCertificate(getKeyPair())
+                .also {
+                    sensitiveStore.store(CERTIFICATE_FILE_NAME, it.serialize())
+                }
+
+    private fun generateKeyPair(): KeyPair {
+        val keyGen = KeyPairGenerator.getInstance(KEY_ALGORITHM)
+        keyGen.initialize(RSA_KEY_MODULUS)
+        return keyGen.generateKeyPair()
+    }
```

Comment:
I'd suggest to use the [`generateRSAKeyPair` function from the Relaynet JVM lib](https://github.com/relaycorp/relaynet-jvm/blob/da8eabbb2335077704d84f27fdf91890053bbba5/src/main/kotlin/tech/relaycorp/relaynet/wrappers/Keys.kt#L16), as it does this exactly, but also makes sure that only RSA keys are generated (using other types of keys in unsupported in Relaynet).